lib/rubygems/source.rb

```diff
@@ -80,7 +80,7 @@ def ==(other) # :nodoc:
   def dependency_resolver_set # :nodoc:
     return Gem::Resolver::IndexSet.new self if 'file' == uri.scheme
 
-    bundler_api_uri = uri + './api/v1/dependencies'
+    bundler_api_uri = uri.merge(uri.path.gsub(/\/+$/, '') + '/').merge('api/v1/dependencies')
```

Comment:
To make the actual fix here more apparent, could we extract this logic to a `enforce_trailing_slash` helper, and keep the previous code untouched otherwise?

```suggestion
    bundler_api_uri = enforce_trailing_slash(uri) + './api/v1/dependencies'
```